content/docs/internals/connection.md

```diff
@@ -0,0 +1,91 @@
+
+:::
+
+- Pomerium presents a server certificate for use based on the provided [`certificates`](/docs/reference/certificates#certificates).
+- Optionally, [client-side mTLS](/docs/capabilities/mtls-clients) can be enabled for downstream connections to enforce the use and validation of client certificates.
+- Pomerium only terminates TLS traffic that meets certain criteria for best practice, which are non-configurable.
+
+### 2. **Request initiation**
+
+After the transport layer connection is established, the downstream client sends an HTTP request. The proxy parses this request, matches it against its configuration, and determines the upstream service to which the request should be forwarded.
+
+### 3. **Request authorization**
+
+The request is authorized according to the policies applied to the Pomerium route before it is forwarded to the upstream service. For more details, see [request lifecycle](/docs/internals/architecture#the-lifecycle-of-a-request).
+
+### 4. **Upstream connection and request forwarding**
+
+- **[Load Balancing](/docs/capabilities/routing#load-balancer)**: Pomerium utilizes its load balancing algorithms to select the optimal upstream service instance to process the request. This selection is made from a list of healthy service instances, which is maintained by service discovery and health checking subsystems. The algorithm used for load balancing can be configured and includes options such as Round Robin, Least Request, Random, and more. See [Load Balancing Policy options](/docs/reference/routes/load-balancing-policy#load-balancing-policy-options) and [Load Balancing Policy Config](/docs/reference/routes/load-balancing-policy-config) for more information.
+- **[Health Checks](/docs/reference/routes/health-checks)**: Health checks are key to maintaining the list of viable service instances for load balancing. Pomerium can be set up to routinely check the health of each upstream service instance using methods like HTTP calls, TCP connections, or custom health checks. If an instance fails the health check, it is deemed unhealthy and removed from the list of available instances for load balancing. This ensures that traffic is not directed to instances incapable of handling it.
+- **Connection & Request Forwarding**: Upon selecting a healthy upstream service instance, Pomerium establishes a connection with this instance (if one does not already exist) and forwards the client's request to it. Connection pooling is employed to improve efficiency by reusing existing connections where possible. This behavior can vary depending on whether the connection is HTTP/1.1 (where each connection can be reused but only manages one request at a time) or HTTP/2 (which supports multiple concurrent requests over a single connection).
```

Comment:
Does this imply that the upstream connection protocol will always match the downstream connection protocol? (i.e. we never proxy an incoming HTTP/1.1 request to an upstream over HTTP/2 or vice versa?)

Reply:
@kenjenkins I'm not sure what's typical for connection pooling, but I agree that this implies the upstream connection protocols must match. I'll follow up with @wasaga.

Reply:
I believe internally Envoy maps HTTP/1 connections to HTTP/2 streams, so it should be possible to proxy an http/1.1 request to http/2 upstream.